Model validation

SupriyaGadigone opened the pull request:
As per discussion in the issue, annotated the model properties + updated the tests. 

The message parameter in the annotations help with finding which tests are broken.

Proof the test pass:
![Screen Shot 2019-04-01 at 2 07 33 PM](https://user-images.githubusercontent.com/15680487/55349737-77943d80-5488-11e9-8b73-01c84e00b1e4.png)

Closes #35

SupriyaGadigone committed:
Model Validation for no null Name's

SupriyaGadigone committed:
Model Validation clean up

RyanWork reviewed (changes requested):
I think it might be better to make the strings `@NotEmpty` or `@NotBlank`. Theoretically, an empty string or whitespace isn't something we want in the database. According to https://www.baeldung.com/javax-validation:

> @NotEmpty – validates that the property is not null or empty; can be applied to String, Collection, Map or Array values

> @NotBlank – can be applied only to text values and validated that the property is not null or whitespace

Do you think there is also a need to write test cases for this? Just to test that our methods would throw exceptions?

SupriyaGadigone committed:
Model Validation tests

SupriyaGadigone commented:
Addressed comments 

Proof the test pass:
![Screen Shot 2019-04-01 at 8 26 30 PM](https://user-images.githubusercontent.com/15680487/55367754-8e548780-54bc-11e9-88e7-1e3d089e3cbb.png)
